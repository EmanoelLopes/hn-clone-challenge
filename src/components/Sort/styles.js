import styled from 'styled-components';

export const SortItems = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-end;
  margin: 1rem 0;

  span {
    margin: 0 1rem;
  }

  button {
    font-size: 16px;
    margin-left: .15rem;
    padding: .5rem .25rem;
  }
`;