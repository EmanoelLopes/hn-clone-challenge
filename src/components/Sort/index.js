import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Button from '../../components/Button';
import { SortItems } from './styles';
import { sortByComments, sortByUpvotes, sortByDate } from '../../store/actions';

function Sort(props) {
  const { sortByComments, sortByUpvotes, sortByDate, visiblePosts } = props;
  return (
    <SortItems>
      <span>Sort By: </span>
      <Button label="Comments" click={() => sortByComments(visiblePosts)}></Button>
      <Button label="Up Votes" click={() => sortByUpvotes(visiblePosts)}></Button>
      <Button label="Date" click={() => sortByDate(visiblePosts)}></Button>
    </SortItems>
  );
};

function mapStateToProps(state) {
  return {
    visiblePosts: state.visiblePosts
  };
};

const mapDispatchToProps = dispatch => ({
  sortByComments: posts => dispatch(sortByComments(posts)),
  sortByUpvotes: posts => dispatch(sortByUpvotes(posts)),
  sortByDate: posts => dispatch(sortByDate(posts)),
})

Sort.propTypes = {
  visiblePosts: PropTypes.array,
  sortByComments: PropTypes.func,
  sortByUpvotes: PropTypes.func,
  sortByDate: PropTypes.func
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Sort);