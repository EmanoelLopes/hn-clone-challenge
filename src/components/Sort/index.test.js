import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import Sort from '../Sort';
import configureStore from '../../store';

const store = configureStore();

configure({ adapter: new Adapter() });

const props = {
  sortByComments: () => {},
  sortByUpvotes: () => {},
  sortByDate: () => {},
  visiblePosts: []
};

describe('<Sort />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Sort {...props} />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('ACTION: SORT_BY_COMMENTS', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Sort {...props} />
      </Provider>
    );
  });

  it('ACTION: SORT_BY_UPVOTES', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Sort {...props} />
      </Provider>
    );
  });

  it('ACTION: SORT_BY_DATE', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Sort {...props} />
      </Provider>
    );
  });
});

