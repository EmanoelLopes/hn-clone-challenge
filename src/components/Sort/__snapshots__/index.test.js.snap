// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sort /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(Sort)
      sortByComments={[Function]}
      sortByDate={[Function]}
      sortByUpvotes={[Function]}
      visiblePosts={Array []}
    />
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "sortByComments": [Function],
      "sortByDate": [Function],
      "sortByUpvotes": [Function],
      "visiblePosts": Array [],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "sortByComments": [Function],
        "sortByDate": [Function],
        "sortByUpvotes": [Function],
        "visiblePosts": Array [],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
